### PR TITLE
New dashboard refactor

### PR DESCRIPTION
**Pull request by heron2014**
Related to #423 

## Note. Merge after #431 
@nelsonic this PR is big because we merged the status and notes. Can you please check it, thank you!

### CODE REVIEW

**Comment by nelsonic** [comment]
this is a *mega* PR! it barely loads on my laptop/browser...

**Comment by nelsonic** [comment]
@SimonLab can you please merge master into this branch/PR so the number of changes are reduced.

**Comment by SimonLab** [comment]
@nelsonic it's merged, I'm not sure if it helps with the number of changes

File: views/partials/navigation.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
       <a class="home-link" href="/clients/list"><i class="fa fa-briefcase"></i></a>
       <a class="home-link" href="/jobs/list"><i class="fa fa-tasks"></i></a>
       <a class="my-favourite" href="/favourite"><i class="fa fa-star"></i></a>
-      <a class="dash-link" href="/dashboard"><i class="fa fa-user"></i></a>
+      <a class="dash-link" href="/gmdashboard"><i class="fa fa-user"></i></a>
```
Comment (iteles) [inline views/partials/navigation.html]: Why did we change this from `dashboard` to `gmdashboard`?

**Comment by nelsonic** [comment]
@SimonLab the PR goes from 3.2k additions down to 1.3k ... its a *lot* easier to review once the commits *already* in `master` are removed. thanks for doing that. 👍

File: assets/css/enhanced.css [hunk 2]
```diff
@@ -259,6 +259,162 @@ span.close > i.fa-times {
 ================================*/
 
 /*==============================
+ new dashboard
+================================*/
+.new-dash-wrapper {
```
Comment (nelsonic) [inline assets/css/enhanced.css]: avoid using the word "*new*" in your class names, its a sign that the project has "*stale*" code...
just get rid of the "old" one.

File: views/partials/navigation.html (hunk 1 above)
Comment (SimonLab) [inline views/partials/navigation.html]: We have build a new endpoint to not delete the old one and we needed to find a new name for it. We can definitely rename this endpoint now we are sure that everything is working well.

Comment (nelsonic) [inline views/partials/navigation.html]: Please re-name before merge so we don't confuse users. 👍

Comment (SimonLab) [inline views/partials/navigation.html]: The endpoint has now a proper name!

**Comment by nelsonic** [comment]
@SimonLab why are you committing `logsHeroku.txt`

**Comment by SimonLab** [comment]
@nelsonic it's a mistake, thanks for spotting this. The logs was used for #440

File: lib/database-helpers/elasticsearch/list_my_clients.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const clientES = require('../../es.js');
+
+// $lab:coverage:off$
```
Comment (nelsonic) [inline lib/database-helpers/elasticsearch/list_my_clients.js]: sorting by name. please add JSDoc comments in the future. 👍

File: lib/database-helpers/elasticsearch/list_my_clients.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+      if (response.hits.total !== numberClients) {
+        clientES.scroll({
+          scrollId: response._scroll_id,
+          scroll: '30s',
+          size: 1000,
+        }, getMoreUntilDone);
+      } else {
+
+        result = result.sort(compare);
```
Comment (nelsonic) [inline lib/database-helpers/elasticsearch/list_my_clients.js]: better to call this function something a bit more *meaningful* e.g: `sort_by_name_asc`

**Comment by nelsonic** [comment]
@SimonLab & @heron2014 the code looks *really* good. 👍 
I would recommend you get into the habit of leaving more communicative comments in your code.
this 1.3k line PR has only 6 "*notes to the next dev*" comments... search for `// ` (*slash slash space*) ...
Happy to merge. but next time ... 😉

**Comment by SimonLab** [comment]
Thanks for the review! I think we should make it mandatory to have some JSDoc on each head of files. We notice that we spend more and more times going through the code to check what a file is doing, with just a few lines of comments we will be more efficient.

**Pull request merged**